src/thunderbots/software/ai/hl/stp/tactic/goalie_tactic.cpp

```diff
@@ -70,65 +126,146 @@ void GoalieTactic::calculateNextIntent(IntentCoroutine::push_type &yield)
+        auto ball_speed_panic =
+            Util::DynamicParameters::GoalieTactic::ball_speed_panic.value();
+        // what should the final goalie speed be, so that the goalie accelarates faster
+        auto goalie_final_speed =
+            Util::DynamicParameters::GoalieTactic::goalie_final_speed.value();
+        // how far in should the goalie wedge itself into the block cone, to block balls
+        auto block_cone_radius =
+            Util::DynamicParameters::GoalieTactic::block_cone_radius.value();
+        // by how much should the defense are be decreased so the goalie stays close
+        // towards the net
+        auto defense_area_deflation =
+            Util::DynamicParameters::GoalieTactic::defense_area_deflation.value();
+
+        // if the ball is in the don't chip rectangle we do not chip the ball
+        // as we risk bumping the ball into our own net trying to move behind
+        // the ball
+        auto dont_chip_rectangle = Rectangle(
+            field.friendlyGoalpostNeg(),
+            field.friendlyGoalpostPos() + Point(2 * ROBOT_MAX_RADIUS_METERS, 0));
+
+        // case 1: goalie should panic and stop the ball, its moving too fast towards the
+        // net
+        if (intersection1.has_value() && ball.velocity().len() > ball_speed_panic)
         {
-            // the ball is heading towards the net. move to block the shot
+            // the ball is heading towards the net, move to intercept the shot
+            // the final speed is a dynamic parameter so that if the goalie needs
+            // to dive for the shot instead of stop when reaching the intersection
+            // point it can do so.
             Point goalie_pos = closestPointOnSeg(
                 (*robot).position(), Segment(ball.position(), *intersection1));
             Angle goalie_orientation = (ball.position() - goalie_pos).orientation();
-
-            next_intent = move_action.updateStateAndGetNextIntent(
-                *robot, goalie_pos, goalie_orientation, 0.0, false, AUTOCHIP);
+            next_intent              = move_action.updateStateAndGetNextIntent(
+                *robot, goalie_pos, goalie_orientation, goalie_final_speed, AUTOCHIP);
         }
-        // Case 2
-        else if (ball.velocity().len() < BALL_SLOW_SPEED_THRESHOLD &&
+        // case 2: goalie does not need to panic and just needs to chip the ball out
+        // of the net
+        else if (ball.velocity().len() <= ball_speed_panic &&
                  field.pointInFriendlyDefenseArea(ball.position()))
         {
-            // if the ball is slow or stationary inside our defense area, chip it out
-            next_intent = chip_action.updateStateAndGetNextIntent(
-                *robot, ball, ball.position(),
-                (ball.position() - field.friendlyGoal()).orientation(), 2);
+            // if the ball is slow but its not safe to chip it out, dont.
+            // TODO finesse the ball out of the goal using the dribbler.
```

Comment:
TODO

Reply:
fixed, removed the TODO as we can't do that yet

Reply:
I would create an issue # for this so we can remember to do it

Reply:
created a ticket and added the comment